java/lang/security/audit/xss/no-direct-response-writer.java

```diff
@@ -191,7 +191,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
         response.setHeader("X-XSS-Protection", "0");
         Object[] obj = {"a", bar};
-        // ruleid: no-direct-response-writer
+        // ruleid: deepok: no-direct-response-writer
```

Comment:
From iago:
```ocaml
  (* Java/tainting:
     * This is a test that comes from the OWASP Benchmark v1.2.
     * Here DeepSemgrep doesn't report lines 56, 128, and 195.
     * But these are actually false positives! This benchmark tries
     * to confuse analyzers into reporting these false positives.
     * It does this in two ways, 1) by using a third-function
     * `doSomething` that receives tainted data, even though it
     * returns safe data; and 2) by putting both safe and unsafe
     * data into a `HashMap`, but ultimately only returning the
     * safe data. FOSS Semgrep falls into the first trap.
     * DeepSemgrep does inter-procedural analysis so it is only
     * affected by the second trap, but it seems to not fall
     * into it because we are lacking a `pattern-propagators` spec
     * for `HashMap`s. If we told DeepSemgrep that `HashMap`s
     * store/propagate taint, then it should report the same
     * false positives.
  *)
```

Reply:
This should probably be a comment in the test here if you're removing it from the overrides in pro. I could see this coming back depending on how sensitive we are to taint in HashMaps.